curriculum/challenges/english/08-coding-interview-prep/rosetta-code/sha-1.english.md

```diff
@@ -0,0 +1,167 @@
+---
+id: 5eb3e4b5f629b9a07429a5d2
+title: SHA-1
+challengeType: 5
+forumTopicId: 385326
+---
+
+## Description
+
+<section id='description'>
+<b>SHA-1</b> or <b>SHA1</b> is a one-way hash function; it computes a 160-bit message digest.
+SHA-1 often appears in security protocols; for example, many HTTPS websites use RSA with SHA-1 to secure their connections.
+BitTorrent uses SHA-1 to verify downloads.
+Git and Mercurial use SHA-1 digests to identify commits.
+A US government standard, <a href="https://rosettacode.org/wiki/SHA-1/FIPS-180-1" target="_blank">FIPS 180-1</a>, defines SHA-1.
+</section>
+
+## Instructions
+
+<section id='instructions'>
+Write a function that returns the SHA-1 message digest for a given string.
+</section>
```

Comment:
I think we should add how SHA-1 is generated and such relevant information.

Reply:
I don't think this is necessary in this case. If we do, the description will be quite long.

The included link toward the bottom of the description breaks things down well. Alternatively, we could also include a link to Wikipedia at the top:

`<a href="https://en.wikipedia.org/wiki/SHA-1" target="_blank"><b>SHA-1</b></a> or <b>SHA1</b> is a one-way hash function; it computes a 160-bit message digest.`

Reply:
Didn't see the link before. That works!